spring-cloud-azure-storage/src/main/java/com/microsoft/azure/spring/cloud/storage/AzureStorageResourcePatternResolver.java

```diff
@@ -0,0 +1,179 @@
+     */
+    public AzureStorageResourcePatternResolver(BlobServiceClient blobServiceClient) {
+        this(blobServiceClient, null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param shareServiceClient the ShareServiceClient.
+     */
+    public AzureStorageResourcePatternResolver(ShareServiceClient shareServiceClient) {
+        this(null, shareServiceClient);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param blobServiceClient the BlobServiceClient.
+     * @param shareServiceClient the ShareServiceClient.
+     */
+    public AzureStorageResourcePatternResolver(
+            BlobServiceClient blobServiceClient, ShareServiceClient shareServiceClient) {
+        this.blobServiceClient = Optional.ofNullable(blobServiceClient);
+        this.shareServiceClient = Optional.ofNullable(shareServiceClient);
+    }
+
+    /**
+     * @see ResourcePatternResolver#getResources(java.lang.String)
+     */
+    @Override
+    public Resource[] getResources(String pattern) throws IOException {
+        Resource[] resources = null;
+
+        if (isAzureStorageResource(pattern, BLOB)) {
+            resources = getBlobResources(pattern);
+        } else if (isAzureStorageResource(pattern, FILE)) {
+            resources = getShareResources(pattern);
+        }
+
+        return resources;
+    }
+
+    /**
+     * @see ResourcePatternResolver#getResource(java.lang.String)
+     */
+    @Override
+    public Resource getResource(String location) {
+        Resource resource = null;
+
+        if (isAzureStorageResource(location, BLOB) && blobServiceClient.isPresent()) {
+            resource = new BlobStorageResource(blobServiceClient.get(), location, true);
+        } else if (isAzureStorageResource(location, FILE) && shareServiceClient.isPresent()) {
+            resource = new FileStorageResource(shareServiceClient.get(), location, true);
+        }
+
+        return resource;
+    }
+
+    /**
+     * @see ResourcePatternResolver#getClassLoader()
+     */
+    @Override
+    public ClassLoader getClassLoader() {
+        return getClass().getClassLoader();
+    }
+
+    /**
+     * Get the blob resources for the given pattern.
+     *
+     * @param pattern the pattern.
+     * @return the blob resources.
+     */
+    private Resource[] getBlobResources(String pattern) {
+        ArrayList<Resource> resources = new ArrayList<>();
+        blobServiceClient.ifPresent(client -> {
+            Iterator<BlobContainerItem> containerIterator
+                    = client.listBlobContainers().iterator();
+            while (containerIterator.hasNext()) {
+                BlobContainerItem containerItem = containerIterator.next();
+                String containerName = containerItem.getName();
+                BlobContainerClient blobContainerClient
+                        = client.getBlobContainerClient(containerItem.getName());
+                Iterator<BlobItem> blobIterator = blobContainerClient.listBlobs().iterator();
```

Comment:
Will this cause a performance issue when using wildcards against a huge blob? Seems like we can't avoid that.

Reply:
Correct it is unavoidable as we have to list the blobs and then determine whether or not it matches against the pattern